apps/gitness/src/pages-v2/repo/repo-summary.tsx

```diff
@@ -47,28 +48,43 @@ export default function RepoSummaryPage() {
     sha: ''
   })
 
-  const { mutate: updateDescription } = useUpdateRepositoryMutation(
+  const [updateError, setUpdateError] = useState<string>('')
+
+  const updateDescription = useUpdateRepositoryMutation(
     { repo_ref: repoRef },
     {
       onSuccess: () => {
+        setUpdateError('')
         refetchRepo()
+      },
+      onError: (error: UpdateRepositoryErrorResponse) => {
+        const errormsg = error?.message || 'An unknown error occurred.'
+        setUpdateError(errormsg)
       }
     }
   )
-  const [isEditingDescription, setIsEditingDescription] = useState<boolean>(false)
-
-  const onChangeDescription = () => {
-    setIsEditingDescription(true)
-  }
 
   const saveDescription = (description: string) => {
-    updateDescription({
+    updateDescription.mutate({
       body: {
         description: description
       }
     })
-    setIsEditingDescription(false)
   }
+  const [submitted, setSubmitted] = useState(false)
+
+  useEffect(() => {
+    if (updateDescription.isSuccess) {
+      setSubmitted(true)
+
+      // After 1 second, reset the submitted state and make the button clickable again
+      const timer = setTimeout(() => {
+        setSubmitted(false)
+      }, 1000)
+
+      return () => clearTimeout(timer)
+    }
+  }, [updateDescription.isSuccess])
```

Comment:
An alternative here would be to base off the `isSuccess` instead of adding additional state. You can then reset the mutation after 1 second. It'll simplify things a little bit and keep the mutation state closer to the UI state

Reply:
sure, sounds good